Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_hotpotato.nut

```diff
@@ -0,0 +1,268 @@
+		foreach ( entity p in GetPlayerArray() )
+		{
+			Highlight_SetEnemyHighlightWithParam1( p, "enemy_sonar", <0, 0, 0> )
+			Remote_CallFunction_NonReplay( p, "ServerCallback_MarkedChanged", victim.GetEncodedEHandle() )
+			NSSendPopUpMessageToPlayer( p, "The potato has been passed to " + victim.GetPlayerName() + "!" )
+		}
+
+		Highlight_SetEnemyHighlight( victim, "enemy_boss_bounty" )
+		SetRoundWinningKillReplayAttacker(attacker)
+		attacker.AddToPlayerGameStat( PGS_ASSAULT_SCORE, 1 )
+		Remote_CallFunction_NonReplay( victim, "ServerCallback_PassedHotPotato" )
+		MarkHasSonar( attacker, victim )
+	}
+}
+
+void function MarkHasSonar( entity old, entity newp ) // give marked the ability to see all players through walls
+{
+	foreach( entity p in GetPlayerArray() )
+	{
+		p.HighlightDisableForTeam( old.GetTeam() )
+		p.HighlightEnableForTeam( newp.GetTeam() )
+	}
+}
+
+void function HotPotatoInitPlayer( entity player )
+{
+	UpdateHotPotatoLoadout( player )
+	Highlight_SetEnemyHighlightWithParam1( player, "enemy_sonar", <0, 0, 0> )
+}
+
+void function HotPotatoPlayerDisconnected( entity player )
+{
+	//if this player was in file.realplayers we remove them from the array as well as lower file.aliveplayers by one
+	if ( file.realplayers.find( player ) != -1 )
+	{
+		file.realplayers.remove( file.realplayers.find( player ) )
+		file.aliveplayers--
+
+		if ( player == file.marked ) // reset countdown etc if marked player suicides or disconnects
+		{
+			foreach ( entity p in GetPlayerArray() )
+			{
+				Highlight_ClearEnemyHighlight(p)
+				Remote_CallFunction_NonReplay( p, "ServerCallback_ShowHotPotatoCountdown", Time() )
+			}
+
+			svGlobal.levelEnt.Signal( "OnMarkedDeath" )
+				
+			thread WaitBeforeMark()
+		}
+	}
+}
+
+void function UpdateHotPotatoLoadout( entity player )
+{
+	if (IsAlive(player) && player != null)
+	{
+		// set loadout
+		foreach ( entity weapon in player.GetMainWeapons() )
+			player.TakeWeaponNow( weapon.GetWeaponClassName() )
+
+		foreach ( entity weapon in player.GetOffhandWeapons() )
+			player.TakeWeaponNow( weapon.GetWeaponClassName() )
+
+		// check if this player is inside file.realplayers
+		if (file.realplayers.find(player) != -1 || !file.firstmarked)
+		{
+			player.GiveOffhandWeapon( "melee_pilot_emptyhanded", OFFHAND_MELEE, [ "allow_as_primary" ])
+			player.GiveOffhandWeapon( "mp_ability_heal", OFFHAND_LEFT )
+			player.SetActiveWeaponByName( "melee_pilot_emptyhanded" )
+
+			SyncedMelee_Disable( player )
+		}
+		else
+		{
+			// let them able to noclip, no collision group as well as be invisible
+			player.SetPhysics( MOVETYPE_NOCLIP )
+			player.kv.VisibilityFlags = 0
+			player.SetTakeDamageType( DAMAGE_NO )
+			player.SetDamageNotifications( false )
+			player.SetNameVisibleToEnemy( false )
+			player.SetNameVisibleToFriendly( false )
+			player.kv.CollisionGroup = TRACE_COLLISION_GROUP_NONE
+
+			Remote_CallFunction_NonReplay( player, "ServerCallback_HotPotatoSpectator" )
+		}
+	}
+}
+

```

Comment:
```suggestion
			Remote_CallFunction_NonReplay( player, "ServerCallback_HotPotatoSpectator" )
		}
		thread RemoveEarnMeter_Threaded( player )
	}
}

void function RemoveEarnMeter_Threaded( entity player )
{
	WaitFrame()
	if ( IsValid( player ) )
		PlayerEarnMeter_SetMode( player, 0 ) // this hides the earn meter UI
}

```
In the interest of polish, since as far as I could tell the earn meter RUI isn't used at all in this mode, we can remove it from players